pyomo/util/tests/test_model_size.py

```diff
@@ -0,0 +1,147 @@
+        model_size = build_model_size_report(eight_process)
+        self.assertEqual(model_size.active.variables, 30)  # 30 is placeholder
+
+    def test_nested_blocks(self):
+        """Test with nested blocks."""
+        m = ConcreteModel()
+        m.b = Block()
+        m.inactive_b = Block()
+        m.inactive_b.deactivate()
+        m.b.x = Var()
+        m.b.x2 = Var(domain=Binary)
+        m.b.x3 = Var(domain=Integers)
+        m.inactive_b.x = Var()
+        m.b.c = Constraint(expr=m.b.x == m.b.x2)
+        m.inactive_b.c = Constraint(expr=m.b.x == 1)
+        m.inactive_b.c2 = Constraint(expr=m.inactive_b.x == 15)
+        model_size = build_model_size_report(m)
+        self.assertEqual(model_size.active.variables, 2)
+        self.assertEqual(model_size.overall.variables, 4)
+        self.assertEqual(model_size.active.binary_variables, 1)
+        self.assertEqual(model_size.overall.binary_variables, 1)
+        self.assertEqual(model_size.active.integer_variables, 0)
+        self.assertEqual(model_size.overall.integer_variables, 1)
+        self.assertEqual(model_size.active.constraints, 1)
+        self.assertEqual(model_size.overall.constraints, 3)
+        self.assertEqual(model_size.active.disjuncts, 0)
+        self.assertEqual(model_size.overall.disjuncts, 0)
+        self.assertEqual(model_size.active.disjunctions, 0)
+        self.assertEqual(model_size.overall.disjunctions, 0)
+
+    def test_disjunctive_model(self):
+        from pyomo.gdp.tests.models import makeNestedDisjunctions
+        m = makeNestedDisjunctions()
+        model_size = build_model_size_report(m)
+        self.assertEqual(model_size.active.variables, 9)
+        self.assertEqual(model_size.overall.variables, 10)
+        self.assertEqual(model_size.active.binary_variables, 7)
+        self.assertEqual(model_size.overall.binary_variables, 7)
+        self.assertEqual(model_size.active.integer_variables, 0)
+        self.assertEqual(model_size.overall.integer_variables, 0)
+        self.assertEqual(model_size.active.constraints, 6)
+        self.assertEqual(model_size.overall.constraints, 6)
+        self.assertEqual(model_size.active.disjuncts, 7)
+        self.assertEqual(model_size.overall.disjuncts, 7)
+        self.assertEqual(model_size.active.disjunctions, 3)
+        self.assertEqual(model_size.overall.disjunctions, 3)
+
+    def test_nonlinear(self):
+        """Test nonlinear constraint detection."""
+        m = ConcreteModel()
+        m.x = Var()
+        m.y = Var()
+        m.z = Var()
+        m.z.fix(3)
+        m.c = Constraint(expr=m.x ** 2 == 4)
+        m.c2 = Constraint(expr=m.x / m.y == 3)
+        m.c3 = Constraint(expr=m.x * m.z == 5)
+        m.c4 = Constraint(expr=m.x * m.y == 5)
+        m.c4.deactivate()
+        model_size = build_model_size_report(m)
+        self.assertEqual(model_size.active.nonlinear_constraints, 2)
+        self.assertEqual(model_size.overall.nonlinear_constraints, 3)
+
```

Comment:
Could you add a test with indexed variables and constraints?

Reply:
Yes, in the process of activating the previously skipped eight-process-problem test.